File: src/main/scala/com/github/bjansen/gatling/blocking/BlockingCode.scala
package com.github.bjansen.gatling.blocking

import com.github.bjansen.gatling.blocking.actions.BlockingActionBuilder
import io.gatling.core.action.builder.ActionBuilder

class BlockingCode(description: String) {

	def run(method: Runnable): ActionBuilder =
		new BlockingActionBuilder(description, method)

}

File: src/main/java/com/github/bjansen/gatling/blocking/javaapi/BlockingCode.java
package com.github.bjansen.gatling.blocking.javaapi;

import com.github.bjansen.gatling.blocking.javaapi.actions.BlockingActionBuilder;

public class BlockingCode {
	private final String description;

	public BlockingCode(String description) {
		this.description = description;
	}
	
	public BlockingActionBuilder run(Runnable runnable) {
		return new BlockingActionBuilder(description, runnable);
	}
}
